### PR TITLE
{bio}[fosscuda/2019b] OpenMM v7.4.1 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
requires ~~#9613~~ (OpenMM patch)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/dcc8b3a9c79a1e22b944c3a645986536 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3307.joltik.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, Python 3.6.8
See https://gist.github.com/a93b5f6d05e5ff5c358eebbbce31d8ee for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in this PR)
node377.hydra.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/7472fde6fa09543a3c09ca5cf9b2c561 for a full test report.

**Comment by smoors** [comment]
@boegel error:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_CUDA_LIBRARY (ADVANCED)
    linked by target "OpenMMCUDA" in directory /tmp/2813100.master01.hydra.brussel.vsc/tmp/vsc10009/build/OpenMM/7.4.1/fosscuda-2019b-Python-3.7.4/openmm-7.4.1/platforms/cuda/sharedTarget
```

**Comment by boegel** [comment]
@smoors Are you buiding on a system with GPUs?

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node254.hydra.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2683 v4 @ 2.10GHz, Python 2.7.5
See https://gist.github.com/f348dfe95b18a206f14d5228a0323b98 for a full test report.

**Comment by smoors** [comment]
@boegel the last report was run on a GPU node. the build succeeds but a lot of tests fail. here are the first failed tests:
```
        Start  31: TestCudaCMAPTorsionForceSingle
 31/284 Test  #31: TestCudaCMAPTorsionForceSingle ............***Failed    4.10 sec
exception: No compatible CUDA device is available

        Start  32: TestCudaCMAPTorsionForceMixed
 32/284 Test  #32: TestCudaCMAPTorsionForceMixed .............***Failed    2.96 sec
exception: No compatible CUDA device is available

        Start  33: TestCudaCMAPTorsionForceDouble
 33/284 Test  #33: TestCudaCMAPTorsionForceDouble ............***Failed    2.95 sec
exception: No compatible CUDA device is available

        Start  34: TestCudaCMMotionRemoverSingle
 34/284 Test  #34: TestCudaCMMotionRemoverSingle .............   Passed    4.31 sec
        Start  35: TestCudaCMMotionRemoverMixed
 35/284 Test  #35: TestCudaCMMotionRemoverMixed ..............   Passed    4.51 sec
        Start  36: TestCudaCMMotionRemoverDouble
 36/284 Test  #36: TestCudaCMMotionRemoverDouble .............   Passed    4.66 sec
        Start  37: TestCudaCheckpointsSingle
 37/284 Test  #37: TestCudaCheckpointsSingle .................***Failed    5.78 sec
exception: The requested CUDA device could not be loaded
```

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/d379d085a851066db90b8a645ff32a31 for a full test report.

**Comment by zao** [comment]
My test report was made on a GTX 1080 (compute capability 6.1), passes with flying colours.

**Comment by boegel** [comment]
> exception: The requested CUDA device could not be loaded

@smoors Any idea why this may happen?

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node253.hydra.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/f8a862761330075040a5a924304cc4af for a full test report.

**Comment by smoors** [comment]
@boegel the problem was that we have 'exclusive process' mode turned on by default.
in my last test, with shared mode, all but one test passed (no clue why):
```
 85/284 Test  #85: TestCudaGayBerneForceSingle ...............***Failed    3.82 sec
exception: Assertion failure at TestGayBerneForce.h:197.  Expected 228.729, found 228.752
```

**Comment by smoors** [comment]
@boegel could you add a note about the 'exclusive process' mode?

**Comment by tardigradus** [comment]
Test report by @tardigradus
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
g014.curta.zedat.fu-berlin.de - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Silver 4210 CPU @ 2.20GHz, Python 2.7.5
See https://gist.github.com/1d9e3132369832429b9c15a977f1a594 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-an03.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, 1 x NVIDIA Quadro K6000, 470.182.03, Python 3.8.10
See https://gist.github.com/akesandgren/2766cf3c6549b3e750c50322ee659f33 for a full test report.

File: easybuild/easyconfigs/o/OpenMM/OpenMM-7.4.1-fosscuda-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    ('CMake', '3.15.3'),
+    ('Doxygen', '1.8.16'),
+]
+
+separate_build_dir = True
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('SciPy-bundle', '2019.10', versionsuffix),
+    ('SWIG', '4.0.1'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/o/OpenMM/OpenMM-7.4.1-fosscuda-2019b-Python-3.7.4.eb:36]: need dep on GL, i.e. libglvnd here
platforms/opencl/src/cl.hpp includes GL/gl.h

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/o/OpenMM/OpenMM-7.4.1-fosscuda-2019b-Python-3.7.4.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/o/OpenMM/OpenMM-7.4.1-fosscuda-2019b-Python-3.7.4.eb:36]: This applies to all `OpenMM` easyconfigs though, no?

Comment (boegel) [reply easybuild/easyconfigs/o/OpenMM/OpenMM-7.4.1-fosscuda-2019b-Python-3.7.4.eb:36]: fixed in bb903f7

if fix is confirmed we can apply same fix to other OpenMM easyconfigs in separate follow-up PR

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
b-an03.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, 1 x NVIDIA Quadro K6000, 470.182.03, Python 3.8.10
See https://gist.github.com/akesandgren/c58db00b29fc8d31c44e814f3b445d3c for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**